frontend/src/app/supportAdmin/ManageUsers/OrgAccessTab.tsx

```diff
@@ -0,0 +1,206 @@
+        (await getTestResultCount({
+          variables: { orgId: user.organization?.id as string },
+          fetchPolicy: "no-cache",
+        }).then((res) => res.data?.testResultsCount)) || 0;
+
+      // if user could lose access of test results stop the submission
+      // and ask for confirmation
+      if (count > 0) {
+        dataRef.current = { ...formData };
+        setShowModal(true);
+        return;
+      }
+    }
+
+    await onSubmit(formData);
+  };
+
+  const handleSubmitFromModal = async () => {
+    closeModal();
+    if (dataRef.current) {
+      await onSubmit(dataRef.current as OrgAccessFormData);
+      dataRef.current = undefined;
+    }
+  };
+
+  /**
+   * Confirmation modal
+   */
+  const [showModal, setShowModal] = useState(false);
+  const closeModal = () => setShowModal(false);
+
+  const confirmationModal = (
+    <Modal
+      onClose={closeModal}
+      title="Organization update"
+      contentLabel="Confirm delete facility"
+      showModal={showModal}
+    >
+      <Modal.Header
+        styleClassNames={"font-heading-lg margin-top-0 margin-bottom-205"}
+      >
+        Organization update
+      </Modal.Header>
+      <div className="border-top border-base-lighter margin-x-neg-205"></div>
+      <p className="margin-top-3">
+        This update will move <span className="text-bold">{fullName}</span> to a
+        different organization. The user will lose access to{" "}
+        <span className="text-bold">{data?.testResultsCount}</span> test{" "}
+        {data?.testResultsCount === 1 ? "result " : "results "}
+        reported under it.
+      </p>
+      <div className="border-top border-base-lighter margin-x-neg-205"></div>
+      <Modal.Footer
+        styleClassNames={"display-flex flex-justify-end margin-top-205"}
+      >
+        <Button
+          className="margin-right-205"
+          variant="unstyled"
+          label="No, go back"
+          onClick={closeModal}
+        />
+        <Button
+          className="margin-right-0"
+          label="Yes, update access"
+          onClick={handleSubmitFromModal}
+        />
+      </Modal.Footer>
+    </Modal>
+  );
+
+  /**
+   * Run validation on facilities when role is changed
+   */
+  useEffect(() => {
+    trigger("facilityIds");
+  }, [formValues.role, trigger]);
+
+  /**
+   * HTML
+   */
+  return (
+    <div
+      role="tabpanel"
+      aria-labelledby={"useraccess-tab"}
+      className="padding-left-1"
+    >
+      <form
+        onSubmit={handleSubmit(handleConfirmationAndSubmit)}
+        className="usa-form usa-form--large manage-user-form__site-admin"
+      >
+        <UserOrganizationFormField
+          control={control}
+          disabled={disabled}
+          isLoadingUser={isLoadingUser}
+        />
+        <UserRoleFormField
+          control={control}
+          registrationProps={register("role", { required: "Role is required" })}
+          error={errors.role}
+          disabled={disabled}
+        />
```

Comment:
I wasn't able to figure out why in the code, but if you load a user that has "Admin" or "Standard" role and then change the role to have lower permissions it doesn't show up on the radio buttons, but the mutation does work. 
For example, load an Admin user and then select "Standard" role or "Testing Only" and it doesn't show in the radio buttons after saving. For a "Standard" user, select "Testing Only".

https://github.com/CDCgov/prime-simplereport/assets/10108172/1dbeac9f-f1f8-46b1-b9c4-09502a8657a9

Reply:
This has bug me for a while and I was hoping to be a "my computer" issue. After many hours of debugging accumulated I think I found the root cause. In the radiogroup component we were setting the `key` props of the radio buttons as random uids... that was somehow messing up the rendering updates. That is why `key` props never should be random 😅  weird things happen.